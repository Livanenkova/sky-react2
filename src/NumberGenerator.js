const arrNumber = [0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,2,25,26,27,28,29,30,31];

function NumberGenerator() {
return (
  <div className="NumberGenerator">
    <div className="NumberGenerator__wrap" style={{ display: "flex", flexWrap: "wrap", width: "350px", gap: "1px", margin: "auto" }}>
      {arrNumber.map( (num) => {
        let backgroundColor = ''
        if(num % 2 === 0) {
          backgroundColor = '#149763'
        } else { 
          for(let i = 2; i < num; i+=1) {
            if(num % i === 0) {
              backgroundColor = '#dafa0a'
              break
            } else {
              backgroundColor = '#bb3d50'
            }
          } 
        }
        return (<div className="NumberGenerator__item" key={num} style={{ width: "40px", height: "40px", background: `${backgroundColor}`, textAlign:"center", margin:"auto" }}>{num}</div>)
      })}  
    </div>
  </div>
)
}

export default NumberGenerator