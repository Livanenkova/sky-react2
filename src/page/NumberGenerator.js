import { Link} from 'react-router-dom'

const arrNumber = [0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31];

function NumberGenerator() {
return (
  <div className="NumberGenerator">
    <div className="NumberGenerator__wrap" style={{ display: "flex", flexWrap: "wrap", width: "350px", gap: "1px", margin: "auto" }}>
      {arrNumber.map( (num) => {
        // Создадим функцию, которая будет проверять, является ли число простым
        const isSimple = (n) => {
          // Если число меньше двух, то оно не может быть простым
          if (n < 2) return false;
          // В цикле делим наше число от 2 до n и проверяем остаток от деления
          for (let i = 2; i < n; i+=1) {
            if (num % i === 0) {
              // Возвращаем false если наше число хоть раз разделилось на число меньше n
              return false;
            }
          }
          // Возвращаем true если наше в предыдущем цикле условие ни разу не выполнилось
          return true;
        }

        let backgroundColor = ''
        // Проверяем является ли число чётным или нечётным, а также проверяем, является ли оно целым
        if (num % 2 === 0) {
          backgroundColor = isSimple(num) ? '#bb3d50' : '#149763'
        } else {
          backgroundColor = isSimple(num) ? '#bb3d50' : '#dafa0a'
        }

        return (<div className="NumberGenerator__item" key={num} style={{ width: "40px", height: "40px", background: `${backgroundColor}`, textAlign:"center", margin:"auto" }}>{num}</div>)
      })}  
    </div>
    <Link to="/">Main Page</Link>
  </div>
)
}

export default NumberGenerator