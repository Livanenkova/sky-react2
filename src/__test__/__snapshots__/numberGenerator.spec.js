/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import React from 'react'
import renderer from 'react-test-renderer';
import {Router } from 'react-router-dom';
import NumberGenerator from '../../page/NumberGenerator';

const { createMemoryHistory } = require("history");


it('отображается корректно', () => {
  const history = createMemoryHistory();
  const tree = renderer.create(<Router location={history.location} navigator={history}><NumberGenerator/></Router>).toJSON();
  expect(tree).toMatchSnapshot();
});
