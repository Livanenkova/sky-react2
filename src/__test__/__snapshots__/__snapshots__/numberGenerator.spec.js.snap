// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`отображается корректно 1`] = `
<div
  className="NumberGenerator"
>
  <div>
    тест
  </div>
  <div
    className="NumberGenerator__wrap"
    style={
      Object {
        "display": "flex",
        "flexWrap": "wrap",
        "gap": "1px",
        "margin": "auto",
        "width": "350px",
      }
    }
  >
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#149763",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      0
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#dafa0a",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      1
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#bb3d50",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      2
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#bb3d50",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      3
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#149763",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      4
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#bb3d50",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      5
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#149763",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      6
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#bb3d50",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      7
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#149763",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      8
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#dafa0a",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      9
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#149763",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      10
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#bb3d50",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      11
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#149763",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      12
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#bb3d50",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      13
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#149763",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      14
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#dafa0a",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      15
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#149763",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      16
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#bb3d50",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      17
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#149763",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      18
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#bb3d50",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      19
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#149763",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      20
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#dafa0a",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      21
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#149763",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      22
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#bb3d50",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      23
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#149763",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      24
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#dafa0a",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      25
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#149763",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      26
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#dafa0a",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      27
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#149763",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      28
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#bb3d50",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      29
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#149763",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      30
    </div>
    <div
      className="NumberGenerator__item"
      style={
        Object {
          "background": "#bb3d50",
          "height": "40px",
          "margin": "auto",
          "textAlign": "center",
          "width": "40px",
        }
      }
    >
      31
    </div>
  </div>
  <a
    href="/"
    onClick={[Function]}
  >
    Main Page
  </a>
</div>
`;
